src/main/java/net/dries007/tfc/client/screen/overlay/PlayerDataOverlay.java

```diff
@@ -0,0 +1,200 @@
+import net.minecraft.potion.Effects;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.MathHelper;
+import net.minecraftforge.api.distmarker.Dist;
+import net.minecraftforge.client.event.RenderGameOverlayEvent;
+import net.minecraftforge.client.gui.ForgeIngameGui;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+
+import com.mojang.blaze3d.matrix.MatrixStack;
+import com.mojang.blaze3d.systems.RenderSystem;
+import net.dries007.tfc.config.TFCConfig;
+import net.dries007.tfc.util.config.HealthDisplayFormat;
+
+import static net.dries007.tfc.TerraFirmaCraft.MOD_ID;
+import static net.minecraftforge.client.event.RenderGameOverlayEvent.ElementType;
+
+@Mod.EventBusSubscriber(bus = Mod.EventBusSubscriber.Bus.FORGE, modid = MOD_ID, value = Dist.CLIENT)
+public final class PlayerDataOverlay
+{
+    private static final ResourceLocation ICONS = new ResourceLocation(MOD_ID, "textures/gui/icons/overlay.png");
+
+    @SuppressWarnings("deprecation")
+    @SubscribeEvent
+    public static void render(RenderGameOverlayEvent.Pre event)
+    {
+        Minecraft mc = Minecraft.getInstance();
+        IngameGui gui = mc.gui;
+        ClientPlayerEntity clientPlayer = mc.player;
+        if (clientPlayer == null) return;
+        PlayerEntity player = clientPlayer.inventory.player;
+
+        ForgeIngameGui.renderFood = TFCConfig.CLIENT.useVanillaHunger.get();
+        ForgeIngameGui.renderHealth = TFCConfig.CLIENT.useVanillaHealth.get();
+        ForgeIngameGui.renderExperiance = TFCConfig.CLIENT.useVanillaHealth.get() && TFCConfig.CLIENT.hideThirstBar.get(); // only allow vanilla exp in this case
+        HealthDisplayFormat healthDisplayFormat = TFCConfig.CLIENT.healthDisplayFormat.get();
+
+        if (event.getType() != ElementType.CROSSHAIRS) return;
+
+        if (!TFCConfig.CLIENT.hideThirstBar.get() || !TFCConfig.CLIENT.useVanillaHunger.get())
+            ForgeIngameGui.right_height += TFCConfig.CLIENT.useVanillaHunger.get() ?6 : 10;
+
+        MatrixStack matrixStack = event.getMatrixStack();
+
+        float displayModifier = (healthDisplayFormat == HealthDisplayFormat.TFC || healthDisplayFormat == HealthDisplayFormat.TFC_CURRENT_HEALTH) ? 50f : 1f;
+        float maxHealth = player.getMaxHealth() * displayModifier; // 20 * 50 = 1000
+        float currentThirst = 100f;
+
+        MainWindow window = event.getWindow();
+        int guiScaledHeight = window.getGuiScaledHeight();
+        int guiScaledWidth = window.getGuiScaledWidth();
+
+        int healthRowHeight = guiScaledHeight - 40;
+        int armorRowHeight = healthRowHeight - 10;
+        int mid = guiScaledWidth / 2;
+
+        FontRenderer fontRenderer = mc.font;
+        RenderSystem.enableBlend();
+        RenderSystem.color4f(1.0f, 1.0f, 1.0f, 1.0f);
+        TextureManager textureManager = mc.getTextureManager();
+        textureManager.bind(ICONS);
+
+        if (!clientPlayer.abilities.instabuild)
```

Comment:
I *think* this should be checking `PlayerEntity#isCreative` rather than `abilities.instabuild`